src/app/ReadHandler.cpp

```diff
@@ -215,35 +174,36 @@ CHIP_ERROR ReadHandler::SendStatusReport(Protocols::InteractionModel::Status aSt
     VerifyOrReturnLogError(IsReportable(), CHIP_ERROR_INCORRECT_STATE);
     if (IsPriming() || IsChunkedReport())
     {
-        mSessionHandle.Grab(mpExchangeCtx->GetSessionHandle());
+        mSessionHandle.Grab(mExchangeCtx->GetSessionHandle());
     }
     else
     {
-        VerifyOrReturnLogError(mpExchangeCtx == nullptr, CHIP_ERROR_INCORRECT_STATE);
+        VerifyOrReturnLogError(mExchangeCtx.Get() == nullptr, CHIP_ERROR_INCORRECT_STATE);
         VerifyOrReturnLogError(mSessionHandle, CHIP_ERROR_INCORRECT_STATE);
-        mpExchangeCtx = InteractionModelEngine::GetInstance()->GetExchangeManager()->NewContext(mSessionHandle.Get().Value(), this);
+        auto exchange = InteractionModelEngine::GetInstance()->GetExchangeManager()->NewContext(mSessionHandle.Get().Value(), this);
+        mExchangeCtx.Grab(exchange);
     }
-    VerifyOrReturnLogError(mpExchangeCtx != nullptr, CHIP_ERROR_INCORRECT_STATE);
 
-    return StatusResponse::Send(aStatus, mpExchangeCtx,
-                                /* aExpectResponse = */ false);
+    VerifyOrReturnLogError(mExchangeCtx.Get() != nullptr, CHIP_ERROR_INCORRECT_STATE);
+    return StatusResponse::Send(aStatus, mExchangeCtx.Get(), /* aExpectResponse = */ false);
 }
 
 CHIP_ERROR ReadHandler::SendReportData(System::PacketBufferHandle && aPayload, bool aMoreChunks)
 {
     VerifyOrReturnLogError(IsReportable(), CHIP_ERROR_INCORRECT_STATE);
     if (IsPriming() || IsChunkedReport())
     {
-        mSessionHandle.Grab(mpExchangeCtx->GetSessionHandle());
+        mSessionHandle.Grab(mExchangeCtx->GetSessionHandle());
     }
     else
     {
-        VerifyOrReturnLogError(mpExchangeCtx == nullptr, CHIP_ERROR_INCORRECT_STATE);
+        VerifyOrReturnLogError(mExchangeCtx.Get() == nullptr, CHIP_ERROR_INCORRECT_STATE);
```

Comment:
All these `mExchangeCtx.Get() == nullptr` could just use the boolean operator on mExchangeCtx, right?

Either way, I guess.